iceoryx_posh/include/iceoryx_posh/popo/event_callback.hpp

```diff
@@ -17,27 +17,58 @@
 #ifndef IOX_POSH_POPO_EVENT_CALLBACK_HPP
 #define IOX_POSH_POPO_EVENT_CALLBACK_HPP
 
+#include "iceoryx_utils/cxx/helplets.hpp"
+
 namespace iox
 {
 namespace popo
 {
+template <typename OriginType, typename ContextDataType>
+struct EventCallback;
+
 namespace internal
 {
 struct NoType_t
 {
 };
+
+using GenericCallbackPtr_t = void (*)();
+using GenericCallbackRef_t = void (&)();
+
+using TranslationCallbackRef_t = void (&)(void* const, void* const, GenericCallbackPtr_t const);
+using TranslationCallbackPtr_t = void (*)(void* const, void* const, GenericCallbackPtr_t const);
+
+template <typename T, typename ContextDataType>
+struct TranslateAndCallTypelessCallback
+{
+    static void call(void* const origin, void* const userType, GenericCallbackPtr_t underlyingCallback)
+    {
+        reinterpret_cast<typename EventCallback<T, ContextDataType>::Ptr_t>(underlyingCallback)(
+            static_cast<T*>(origin), static_cast<ContextDataType*>(userType));
+    }
+};
+
+template <typename T>
+struct TranslateAndCallTypelessCallback<T, internal::NoType_t>
+{
+    static void call(void* const origin, void* const userType, GenericCallbackPtr_t underlyingCallback)
+    {
+        IOX_DISCARD_RESULT(userType);
```

Comment:
`[[gnu::unused]]`?

Reply:
I prefer `IOX_DISCARD_RESULT` since it does not cause warnings on windows since it is a gnu extension.

Reply:
Ok, just wondering if we should introduce and alias aka `IOX_INTENTIONALLY_NOT_USED` or something like that

Reply:
I think we need a `IOX_MAYBE_UNUSED` when we fully support Windows in order to replace all occurrences of `[[gnu::unused]]`